libd/src/d/semantic/dtemplate.d

```diff
@@ -159,6 +159,26 @@ private:
 				return false;
 			}
 		}
+				
+		if(t.constraint) {
+			
+			auto oldScope = currentScope;
+			scope(exit) {currentScope = oldScope;}
+			
+			auto ti = instanciateFromResolvedArgs(t.location, t, matchedArgs, true);
```

Comment:
Why do you instantiate BEFORE checking the constraint ? This is going to be awfully slow and memory hungry.